api-participant/src/main/java/bio/terra/pearl/api/participant/config/B2CConfigurationService.java

```diff
@@ -0,0 +1,99 @@
+package bio.terra.pearl.api.participant.config;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.InputStream;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.stereotype.Service;
+import org.yaml.snakeyaml.LoaderOptions;
+import org.yaml.snakeyaml.Yaml;
+import org.yaml.snakeyaml.constructor.Constructor;
+
+@Slf4j
+@Service
+public class B2CConfigurationService {
+
+  @Getter private B2CConfiguration b2cConfiguration;
+  private final Map<String, Map<String, String>> portalToConfig = new HashMap<>();
+
+  public B2CConfigurationService() {
+    b2cConfiguration = new B2CConfiguration();
+  }
+
+  /**
+   * Get the B2C configuration for a portal. Returns an empty map if portal config for portal is not
+   * specified
+   */
+  public Map<String, String> getB2CForPortal(String portalShortcode) {
+    // see if already constructed
```

Comment:
```suggestion
    /** if it's already constructed use that to save the expense of recomputing.  Otherwise, try rereading the config in case new values have been propagated */
```
Is the reason we're not just reading the config into the string maps all at once at startup to better pick up dynamic changes?

Reply:
Lazy initialization, for no particular reason. Can change if it makes things clearer.
Not sure what you mean about re-reading the config. What would tigger that?

Reply:
I expected the map to be read all at once.  I might lean towards that because it will pick up any config parsing errors immediately, as opposed to waiting until the first request comes in.  But I could see it either way

Reply:
That makes sense. Will change.